aws-sagemaker-hosted-scorer/src/main/java/ai/h2o/mojos/deploy/sagemaker/hosted/controller/ModelsApiController.java

```diff
@@ -0,0 +1,108 @@
+        return this.isReady;
+    }
+
+    /**
+     * Simple Api controller. Inherits from {@link ModelApi}, which controls global,
+     * expected request mappings for the rest service.
+     *
+     * Sagemaker Specific: override `getScore` method to point to requestMapping `/invocations`
+     *                     add `ping` method that points to `/ping` for Sagemaker health checks
+     * See Sagemaker Docs here:
+     * https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-inference-code.html
+     * @param scorer
+     * {@link MojoScorer} initialized class containing loaded mojo, and mojo interaction methods
+     * @param sampleRequestBuilder
+     * {@link SampleRequestBuilder} Simple class for generating sample request.
+     */
+    @Autowired
+    public ModelsApiController(MojoScorer scorer, SampleRequestBuilder sampleRequestBuilder) {
+        this.scorer = scorer;
+        log.info("Mojo pipeline successfully loaded ({}).", scorer.getModelId());
+        this.sampleRequestBuilder = sampleRequestBuilder;
+        this.isReady = true;
+    }
+
+    @Override
+    public ResponseEntity<Model> getModelInfo() {
+        return ResponseEntity.ok(scorer.getModelInfo());
+    }
+
+    @Override
+    public ResponseEntity<String> getModelId() {
+        return ResponseEntity.ok(scorer.getModelId());
+    }
+
+    @Override
+    @RequestMapping("/invocations")
+    public ResponseEntity<ScoreResponse> getScore(ScoreRequest request) {
+        try {
+            log.info("Got scoring request");
+            return ResponseEntity.ok(scorer.score(request));
+        } catch (Exception e) {
+            log.info("Failed scoring request: {}, due to: {}", request, e.getMessage());
+            log.debug(" - failure cause: ", e);
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @Override
+    public ResponseEntity<ScoreResponse> getScoreByFile(String file) {
+        if (Strings.isNullOrEmpty(file)) {
+            log.info("Request is missing a valid CSV file path");
+            return ResponseEntity.badRequest().build();
+        }
+        try {
+            log.info("Got scoring request for CSV");
+            return ResponseEntity.ok(scorer.scoreCsv(file));
+        } catch (IOException e) {
+            log.info("Failed loading CSV file: {}, due to: {}", file, e.getMessage());
+            log.debug(" - failure cause: ", e);
+            return ResponseEntity.badRequest().build();
+        } catch (Exception e) {
+            log.info("Failed scoring CSV file: {}, due to: {}", file, e.getMessage());
+            log.debug(" - failure cause: ", e);
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @Override
+    public ResponseEntity<ScoreRequest> getSampleRequest() {
+        return ResponseEntity.ok(sampleRequestBuilder.build(scorer.getPipeline().getInputMeta()));
+    }
+
+    @RequestMapping("/ping")
+    public ResponseEntity<String> ping() {
```

Comment:
This seems to always return `ok`. I would tend to remove both the field `isReady` and the getter `getReady` and only return a constant `ok` here.

Reply:
:+1: